### PR TITLE
Cleanup permissions checking methods

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Move permission checking methods from mixins to a separate file and update all the places that use them.

## Related issues

Needed for #4219

### CODE REVIEW

File: jsapp/js/popoverMenu.tsx [hunk 1]
```diff
@@ -13,9 +13,9 @@ interface PopoverMenuProps {
   type?: string;
   additionalModifiers?: string[];
   /** the element that will be opening the menu, menu will be placed in relation to it */
-  triggerLabel: JSX.Element;
+  triggerLabel: React.ReactNode;
```
Comment (p2edwards) [inline jsapp/js/popoverMenu.tsx:16]: Interesting. Should we apply this to helpBubble.tsx as well? 

```patch
diff --git a/jsapp/js/components/support/helpBubble.tsx b/jsapp/js/components/support/helpBubble.tsx
index 295c7437a..cd214813a 100644
--- a/jsapp/js/components/support/helpBubble.tsx
+++ b/jsapp/js/components/support/helpBubble.tsx
@@ -277,7 +277,7 @@ class HelpBubble extends React.Component<{}, HelpBubbleState> {
   }
 
   render() {
-    let popupRenderFn: () => JSX.Element | null = () => null;
+    let popupRenderFn: () => React.ReactNode | null = () => null;
     const modifiers = ['support'];
     if (this.state.isOpen) {
       modifiers.push('open');
```

**Review by p2edwards — CHANGES_REQUESTED** [review]
Glad to see mixins.tsx getting smaller. :)

File: jsapp/js/components/submissions/tableSettings.es6 [hunk 2]
```diff
@@ -133,7 +131,4 @@ class TableSettings extends React.Component {
   }
 }
 
-reactMixin(TableSettings.prototype, Reflux.ListenerMixin);
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/submissions/tableSettings.es6:136]: This `Reflux.ListenerMixin` is responsible for `this.listenTo(…)` method. It's no longer used here, hence the cleanup :)

File: jsapp/js/components/formViewTabs.es6 [hunk 3]
```diff
@@ -12,6 +12,7 @@ import {PERMISSIONS_CODENAMES} from 'js/constants';
 import {ROUTES} from 'js/router/routerConstants';
 import {withRouter} from 'js/router/legacy';
 import {assign} from 'utils';
+import {userCan} from 'js/components/permissions/utils';
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/formViewTabs.es6:15]: ✅

File: jsapp/js/components/submissions/tableBulkOptions.es6 [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
 import PopoverMenu from 'js/popoverMenu';
-import mixins from 'js/mixins';
 import alertify from 'alertifyjs';
 import {
   MODAL_TYPES,
   VALIDATION_STATUSES_LIST,
   PERMISSIONS_CODENAMES,
 } from 'js/constants';
 import {renderCheckbox} from 'utils';
+import {userCan} from 'js/components/permissions/utils';
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/submissions/tableBulkOptions.es6:14]: ✅

File: jsapp/js/components/submissions/submissionModal.es6 [hunk 5]
```diff
@@ -20,6 +19,8 @@ import {
 } from 'js/constants';
 import SubmissionDataTable from 'js/components/submissions/submissionDataTable';
 import Checkbox from 'js/components/common/checkbox';
+import {isSubmissionWritable} from 'js/components/permissions/utils';
+import {userCan} from 'js/components/permissions/utils';
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/submissions/submissionModal.es6:23]: ✅

File: jsapp/js/popoverMenu.tsx (hunk 1 above)
Comment (magicznyleszek) [reply jsapp/js/popoverMenu.tsx:16]: `React.ReactNode` and `JSX.Element` are very similar. And I've learned more about their differences: https://stackoverflow.com/a/72353143/2311247. I think that unless TS complains, there is no need to change these.

**Pull request merged**